### PR TITLE
[cinder] Migrate policy.json to policy.yaml

### PR DESCRIPTION
**Pull request by hemna**
For cinder Queens and on, Cinder can use a policy.yaml file
instead of a policy.json file.  The policy.yaml is the default
for Rocky release and forward.

The rocky release notes describes the upgrade.
https://docs.openstack.org/releasenotes/cinder/rocky.html

### CODE REVIEW

**Comment by hemna** [comment]
This patch is a pre-emptive migration for Train upgrade.

**Review by joker-official — COMMENTED** [review]
Did you manually compare the rules and copy over our stuff to yaml or how did you convert the file?

**Comment by hemna** [comment]
As described in the release notes, the yaml file is basically. identical minus the wrapping {} around the json, and removing
of the , at the end of each line.   

So I basically did this:
cp policy.json policy.yaml
edit the file, removed the {} and , 

Then I generated the default cinder-policy.yaml.sample file from tox -egenpolicy
and compared the policy entries.  I changed some of the invalid policy entries that we had for
managing volume attachments, such as 
"volume:attach" : "...."  to the correct entry using the volume_extension actions.
I also added the group actions which are replacements for the old consistency group entries as CG's as an API don't exist anymore.

File: openstack/cinder/templates/etc/_cinder-policy.yaml.tpl [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+"owner": "project_id:%(project_id)s"
+"member": "role:member and rule:owner"
+"viewer": "role:volume_viewer and rule:owner"
+"admin": "role:volume_admin and rule:owner"
+"context_is_volume_admin": "rule:context_is_admin or rule:admin"
+"context_is_editor": "rule:context_is_volume_admin or rule:member"
+"context_is_viewer":  "rule:context_is_editor or rule:viewer"
+"view_all": "role:member or role:volume_viewer or role:volume_admin or role:cloud_volume_admin"
+"edit_all": "role:member or role:volume_admin or role:cloud_volume_admin"
+"default": "rule:context_is_admin"
```
Comment (joker-official) [inline openstack/cinder/templates/etc/_cinder-policy.yaml.tpl:19]: Release-Notes state, that `default` is not used inside the code anymore, if I read this correctly.

File: openstack/cinder/templates/etc/_cinder-policy.yaml.tpl [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+"group:update_group_snapshot" :   "rule:context_is_admin"
+"group:group_snapshot_project_attribute" :   "rule:context_is_admin"
+"group:reset_group_snapshot_status" :   "rule:context_is_admin"
+"group:enable_replication" :   "rule:context_is_admin"
+"group:disable_replication" :   "rule:context_is_admin"
+"group:failover_replication" :   "rule:context_is_admin"
+"group:list_replication_targets" :   "rule:context_is_admin"
+
+# List all backend pools
+"scheduler_extension:scheduler_stats:get_pools" : "rule:context_is_admin
```
Comment (joker-official) [inline openstack/cinder/templates/etc/_cinder-policy.yaml.tpl:143]: This line is missing a closing `"`.

**Review by joker-official — CHANGES_REQUESTED** [review]

**Comment by joker-official** [comment]
Nice explanation, thanks.

File: openstack/cinder/templates/etc/_cinder-policy.yaml.tpl [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+"group:create_group_snapshot" :   "rule:context_is_admin"
+"group:get_group_snapshot" :   "rule:context_is_admin"
+"group:delete_group_snapshot" :   "rule:context_is_admin"
+"group:update_group_snapshot" :   "rule:context_is_admin"
+"group:group_snapshot_project_attribute" :   "rule:context_is_admin"
+"group:reset_group_snapshot_status" :   "rule:context_is_admin"
+"group:enable_replication" :   "rule:context_is_admin"
+"group:disable_replication" :   "rule:context_is_admin"
+"group:failover_replication" :   "rule:context_is_admin"
+"group:list_replication_targets" :   "rule:context_is_admin"
```
Comment (joker-official) [inline openstack/cinder/templates/etc/_cinder-policy.yaml.tpl:139]: hm ... I'm annoyed by those lines having a space before the `:`. We don't do it everywhere in the file. Can we straighten this out with a small run of `sed -r 's/" +:/":/' _cinder-policy.yaml.tpl`, please?

**Review by joker-official — APPROVED** [review]

**Pull request merged**